src/customloaderinterface.hpp

```diff
@@ -0,0 +1,108 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#ifndef SRC_CUSTOMLOADERINTERFACE_HPP_
+#define SRC_CUSTOMLOADERINTERFACE_HPP_
+
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace ovms {
+
+/**
+     * @brief This class is the custom loader interface base class.
+     * Custom Loaders need to implement this interface and define the virtual functions to enable
+     * OVMS load a model using a custom loader
+     */
+class CustomLoaderInterface {
+public:
+    /**
+         * @brief Constructor
+         */
+    CustomLoaderInterface() {
+        printf("Created custom loader object................\n");
+    }
+    /**
+         * @brief Destructor
+         */
+    virtual ~CustomLoaderInterface() {
+        printf("Deleted custom loader object................\n");
+    }
+
+    /**
+         * @brief Initialize the custom loader
+         *
+         * @param loader config file defined under custom loader config in the config file
+         *
+         * @return status
+         */
+    virtual int loaderInit(char* loaderConfigFile) = 0;
+
+    /**
+         * @brief Load the model by the custom loader
+         *
+         * @param model name required to be loaded - defined under model config in the config file
+         * @param base path where the required IR files are present
+         * @param version of the model
+         * @param loader config parameters json as string
+         * @param char pointer to the model xml buffer
+         * @param length of the model xml buffer
+         * @param char pointer to the weights buffer
+         * @param length of the weights buffer
+         * @return status
+         */
+    virtual int loadModel(const char* modelName,
+        const char* basePath,
+        const int version,
+        const char* loaderOptions,
+        char** xmlBuffer, int* xmlLen,
+        char** binBuffer, int* binLen) = 0;
+
+    /**
+         * @brief Get the model black list status

```

Comment:
This is one of the ways for customloader to tell model server that this model is not good for loading. This will be useful if customer loader is implementing some sort of licensing for the model and once the license expires, the model can be blacklisted till a new license is obtained. Once new license is obtained, custom loader can remove the model from black list